Rest/Controller/ContentController.php

```diff
@@ -0,0 +1,229 @@
+        $this->locationService = $locationService;
+        $this->contentTypeService = $contentTypeService;
+        $this->searchService = $searchService;
+    }
+
+    /**
+     * Prepares content for ContentDataValue class.
+     *
+     * @param string $contentIdList
+     *
+     * @return \EzSystems\RecommendationBundle\Rest\Values\ContentData
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException     if the content, version with the given id and languages or content type does not exist
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the user has no access to read content and in case of un-published content: read versions
+     */
+    public function getContent($contentIdList)
+    {
+        $contentIds = explode(',', $contentIdList);
+        $content = $this->prepareContent($contentIds);
+
+        return new ContentDataValue($content);
+    }
+
+    /**
+     * Prepare content array.
+     *
+     * @param array $contentIds
+     *
+     * @return array
+     */
+    protected function prepareContent($contentIds)
+    {
+        $requestLanguage = $this->request->get('lang');
+        $requestedFields = $this->request->get('fields');
+
+        $content = array();
+
+        foreach ($contentIds as $contentId) {
+            $contentValue = $this->contentService->loadContent($contentId, [$requestLanguage]);
+            $contentType = $this->contentTypeService->loadContentType($contentValue->contentInfo->contentTypeId);
+            $location = $this->locationService->loadLocation($contentValue->contentInfo->mainLocationId);
+            $language = (is_null($requestLanguage)) ? $contentType->mainLanguageCode : $requestLanguage;
+
+            $content[$contentId] = array(
+                'contentId' => $contentId,
+                'contentTypeId' => $contentType->id,
+                'identifier' => $contentType->identifier,
+                'language' => $language,
+                'publishedDate' => $contentValue->contentInfo->publishedDate->format('c'),
+                'author' => $contentValue->getFieldValue('author'),
+                'uri' => $this->generator->generate($location, array(), false),
+                'mainLocation' => array(
+                    'href' => '/api/ezp/v2/content/locations'.$location->pathString,
+                ),
+                'locations' => array(
+                    'href' => '/api/ezp/v2/content/objects/'.$contentId.'/locations',
+                ),
+                'categoryPath' => $location->pathString,
+            );
+
+            $fields = $this->prepareFields($contentType, $requestedFields);
+
+            if (!empty($fields)) {
+                foreach ($fields as $field) {
+                    $fieldValue = $contentValue->getFieldValue($field, $language);
+
+                    if (null === $fieldValue) {
+                        continue;
+                    } elseif (key($fieldValue) == 'destinationContentId') {
```

Comment:
I would also check if field type is relation, and if so, then look for destionationContentId